godot-core/src/init/mod.rs

```diff
@@ -53,17 +55,38 @@ pub unsafe fn __gdext_load_library<E: ExtensionLibrary>(
     is_success.unwrap_or(0)
 }
 
+static LEVEL_SERVERS_CORE_LOADED: AtomicBool = AtomicBool::new(false);
+
 unsafe extern "C" fn ffi_initialize_layer<E: ExtensionLibrary>(
     _userdata: *mut std::ffi::c_void,
     init_level: sys::GDExtensionInitializationLevel,
 ) {
     let level = InitLevel::from_sys(init_level);
     let ctx = || format!("failed to initialize GDExtension level `{:?}`", level);
 
-    // Swallow panics. TODO consider crashing if gdext init fails.
-    let _ = crate::private::handle_panic(ctx, || {
+    fn try_load<E: ExtensionLibrary>(level: InitLevel) {
+        // Workaround for https://github.com/godot-rust/gdext/issues/629:
+        // When using editor plugins, Godot may unload all levels but only reload from Scene upward.
+        // Manually run initialization of lower levels.
+
+        // TODO: Remove this workaround once after the upstream issue is resolved.
+        if level == InitLevel::Scene {
+            if !LEVEL_SERVERS_CORE_LOADED.load(Relaxed) {
+                try_load::<E>(InitLevel::Core);
+                try_load::<E>(InitLevel::Servers);
+            }
+        } else if level == InitLevel::Servers {
+            // When it's normal initialization, the `Servers` level is normally initialized.
+            LEVEL_SERVERS_CORE_LOADED.store(true, Relaxed);
+        }
```

Comment:
There's an asymmetry with the unloading below, which is set in `Core`, and here it's `Servers` level.

To be conservative, I'd probably only set this one in `Core` as well (otherwise `LEVEL_SERVERS_CORE_LOADED` is true while `Core` is not yet loaded/loading).

Reply:
Applied in new commit. Thanks!